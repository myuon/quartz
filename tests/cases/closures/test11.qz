fun js_fetch(url: string, call: fun[(i32), nil]) {
    call(10);
}

fun js_image_set_onload(image: any, call: fun[(), nil]) {
    call();
}

fun js_image_set_onerror(image: any, call: fun[(), nil]) {
    call();
}

fun js_context_draw_image(context: string, image: i32, x: i32, y: i32) {
    println(context);
    println("{}", image.to_string());
    println("{}", x.to_string());
    println("{}", y.to_string());
}

fun main() {
    let context = "context";
    let image = "image";

    js_fetch(
        "assets/rhb.json",
        fun (resp_id: i32) {
            let rhb = 200;
            js_image_set_onload(
                rhb,
                fun () {
                    js_context_draw_image(context, rhb, 0, 0);
                },
            );
            js_image_set_onerror(
                image,
                fun () {
                    println("Image failed to load");
                },
            );
        },
    );

    println("ok");
}

