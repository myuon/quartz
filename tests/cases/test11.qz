struct Point {
    x: i32,
    y: i32,
}

module Point {
    fun get_x(self): i32 {
        return self.x;
    }

    fun sum(self): i32 {
        return self.get_x() + self.y;
    }

    fun new(x: i32, y: i32): Point {
        return Point {
            x: x,
            y: y,
        };
    }
}

fun main(): i32 {
    let p = Point::new(10, 20);

    return p.sum();
}

