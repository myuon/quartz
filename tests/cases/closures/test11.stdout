context
200
0
0
Image failed to load
ok
